### PR TITLE
Add sample code for Dataflow reading Spanner table with Index

### PR DESCRIPTION
**Pull request by nielm**
- [X] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ N/A ] `pom.xml` parent set to latest `shared-configuration`
- [ N/A ] Appropriate changes to README are included in PR
- [ None ] API's need to be enabled to test (tell us)
- [ None ] Environment Variables need to be set (ask us to set them)
- [ X ] **Tests** pass:   `mvn clean verify` **required**
- [ X ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ X ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
   -  ignoring USBR_UNNECESSARY_STORE_BEFORE_RETURN in new file for code sample.
- [ X ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: dataflow/spanner-io/src/main/java/com/example/dataflow/SpannerReadApiWithIndex.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2017 Google Inc.
```
Comment (Shabirmean) [inline dataflow/spanner-io/src/main/java/com/example/dataflow/SpannerReadApiWithIndex.java:2]: Update copyright to 2023

**Review by Shabirmean — APPROVED** [review]
LGTM!

Please resolve the 1 comment. Sorry for the delay, I was away on vacation and just got back last night!

File: dataflow/spanner-io/src/main/java/com/example/dataflow/SpannerReadApiWithIndex.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    @Description("Output filename for records size")
+    @Validation.Required
+    String getOutput();
+
+    void setOutput(String value);
+  }
+
+  public static void main(String[] args) {
+    Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
```
Comment (Shabirmean) [inline dataflow/spanner-io/src/main/java/com/example/dataflow/SpannerReadApiWithIndex.java:69]: Can you give an example of what the args look like here?

Comment (nielm) [reply dataflow/spanner-io/src/main/java/com/example/dataflow/SpannerReadApiWithIndex.java:69]: These are standard args for a dataflow job, documented below in Options class, and in the DataflowOptions class. javadoc added.

(this file was forked from SpannerReadApi.java)

File: dataflow/spanner-io/src/main/java/com/example/dataflow/SpannerReadApiWithIndex.java (hunk 1 above)
Comment (nielm) [reply dataflow/spanner-io/src/main/java/com/example/dataflow/SpannerReadApiWithIndex.java]: Done.

File: dataflow/spanner-io/src/main/java/com/example/dataflow/SpannerReadApiWithIndex.java (hunk 2 above)
Comment (nielm) [reply dataflow/spanner-io/src/main/java/com/example/dataflow/SpannerReadApiWithIndex.java:72]: Added javadoc to link to Options class. 

These are standard Dataflow args, This file was forked from SpannerReadApiWithIndex.java

**Pull request merged**